### PR TITLE
docs: add React Native Standards

### PR DESCRIPTION
**Pull request by tagadiadmt**
## What does this PR do?
- Introduces a comprehensive overview of best practices, coding standards, and recommendations for React Native development

## Why did you choose this solution?
- To help developers write clean, efficient, and maintainable code

## What is the dependency graph of the changes?
- N/A

## How to test/verify?
- Review the standards document for completeness and adherence to best practices

## Related ticket url:
-

### CODE REVIEW

**Comment by symphdevtools** [comment]
Hi! I'm generating PR title and description for you and will automatically apply it to your PR. I'll react :rocket: to this comment when done. Just sit back and chill!

When you have new changes to this PR and you want to regenerate the title and PR description, just comment `churva`!

Let me know if something is not working well, e.g.:
- Some parts of the title and description weren't preserved when generating title and description
- Disappearing screenshots or links
- Etc.

Thanks!

(RIP hatdog)

File: standards/react_native_standards.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
+# Style Guide
+
+This style guide provides a comprehensive overview of best practices, coding standards, and recommendations for working with React Native. It aims to help developers write clean, efficient, and maintainable code.
+
+## General Principles
+
+- **Component-Based Development**: Break down your UI into reusable components, each responsible for a specific piece of functionality or view. This approach promotes code reusability, maintainability, and scalability.
```
Comment (JHNLWHD) [inline standards/react_native_standards.md:7]: Let's confirm with our RN devs what do we follow, Is it atomic-based or feature-based?

File: standards/react_native_standards.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+- **State Management**: Keep state as localized as possible to avoid unnecessary re-renders and maintain a clear data flow.
```
Comment (JHNLWHD) [inline standards/react_native_standards.md:14]: Let's mention also global state management such as user data

File: standards/react_native_standards.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+- **Performance Optimization**: Minimize unnecessary re-renders, efficiently render large lists of data, and profile and optimize performance bottlenecks using react tools.
+- **Platform-Specific Considerations**: Utilize platform-specific components or styles when necessary to accommodate differences between iOS and Android.
+- **Testing and Debugging**: Write comprehensive tests to ensure the reliability and stability of your React Native application.
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: Let's mention what testing library we prefer

File: standards/react_native_standards.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+
+## Basic Rules
+
+- Always use `const` or `let` to declare variables. Use const by default, unless a variable needs to be reassigned.
+- Always use functional component.
+- No nested components or getComponent inside render method. Always try to separate the components if possible.
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: Add sample codes

File: standards/react_native_standards.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+  > `myUtilityFile.js` <br>
+  > `cookieHelper.js` <br>
+  > `fetchApi.js`
+
+- Unit test files should use the same name as its corresponding file.
+  > `HeroBanner.js` <br>
+  > `HeroBanner.test.js` <br>
+  
+  > `CookieBanner.js` <br>
+  > `CookieBanner.test.js`
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: Mention if we prefer a separate test folder or colocation

File: standards/react_native_standards.md [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+
+- Global StyleSheet file should be written using `camel case`.
+  > `globalStyle.js` <br>
+
+- All fonts, colors, and font sizes should be in one file.
+  > `fontsConfig.js`
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: Samples should be in `.ts`

File: standards/react_native_standards.md [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+- Attribute name should be `camel case`.
+  > `className` <br>
+  > `onClick`
+
+- Variable names should be `camel case`. Variable names can contain number and special characters.
+  > `const variable = 'test';` <br>
+  > `let variableBoolean = true;`
+
+- Use `.jsx` or `.tsx` extension a for React components
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: We prefer `.tsx`

File: standards/react_native_standards.md [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+  ```csharp
+  import styles from'./myComponent.styles';
+  import image from './image.png';
+  ```
+
+- Test-Only Imports
+
+  ```csharp
+  import { testFunction } from './tests';
+  ```
```
Comment (JHNLWHD) [inline standards/react_native_standards.md:188]: Mention the specific eslint-config responsible for this, this should be automated

File: standards/react_native_standards.md [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+## Writing with TypeScript
+
+- To improve code readability, TypeScript comes into play. TypeScript adds types to JavaScript, enabling us to specify the type of each property in the object, reducing the possibility of errors. With TypeScript, you and your editor are aware of the object properties and their types in the code, making it easier to understand and extend the functionality.
+
+## Path Aliases
+
+- Using path aliases is giving your code a shortcut to make import paths shorter and more understandable. This is especially handy when you have many folders nested inside each other, making your imports easier to read.
+
+## Usage of Double Quotes (" ") & Single Quotes (' ')
+- To avoide the confusion about double quotes (" ") & single quotes (' '), use double quotes (" ") for JSX attributes and single quotes (' ') for the JS code
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: For this we can mention the eslint and prettier configs

File: standards/react_native_standards.md [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+
+## Crash Reporting Tools
+
+- Crash analytics tools are like superhero sidekicks for your app, always watching for any trouble 24/7. They quickly catch and understand app crashes and errors in real-time, helping you figure out what went wrong.
+- Some cool crash analytics tools are `Sentry`, `Firebase`, `Crashlytics`, and more
```
Comment (JHNLWHD) [inline standards/react_native_standards.md]: Let's sync up with the team what's our default standard for this, I know it's Firebase Crashlytics

**Review by JHNLWHD — CHANGES_REQUESTED** [review]

**Review by JHNLWHD — APPROVED** [review]
LGTM 👍🏼

**Pull request merged**